### DISCUSSION TITLE
Cam shows an error while detection

### DISCUSSION THREAD

**Discussion by Zeynp62** [ultralytics/ultralytics#6549] [Q&A] [answered]
: [ WARN:1@51.468] global cap_msmf.cpp:1759 CvCapture_MSMF::grabFrame videoio(MSMF): can't grab frame. Error: -1072873821

I'm working with custom detection model connected to Arduino Uno. While running the code the above error shows. What is the reason? and what should I do to solve the problem?

**Comment by yonglee7015** [comment, marked as answer]
@Zeynp62, it appears there might be an issue with OpenCV and your camera. The error code `-1072873821` corresponds to `MF_E_HW_MFT_FAILED_START_STREAMING`, indicating that a hardware device failed to initiate streaming. This could occur if the camera is already in use by another application or if there is a malfunction with the camera device.

Here are some potential solutions:
✅ Verify that your camera functions correctly with other applications, such as the Windows Camera app.
✅ Prior to opening a new device, attempt to release the camera using `capture.release()`.
✅ Instead of a fixed number of iterations, employ an infinite loop in a thread, and confirm that `capture.read()` returns `True`.

I hope this helps you to fix the error. If you need more information, you can check out the web search results below. 🙌

### Web search findings
✅ [OpenCV CvCapture_MSMF::grabFrame videoio(MSMF): can't grab frame](https://stackoverflow.com/questions/63156943/opencv-cvcapture-msmfgrabframe-videoiomsmf-cant-grab-frame-error-214748)
✅ [Cv2 CvCapture_MSMF::grabFrame videoio(MSMF): can't grab frame](https://stackoverflow.com/questions/53219618/opencv-java-warn2-videoiomsmf-cant-grab-frame-error-1072875772)

Hope this helps! Let me know if you have any additional questions.
Thanks

**Reply by Zeynp62** [reply]
I'm using this line of code to start the camera with yolov8 detection model :         results = model(source=0, show=True, conf=0.5, save=False, classes=[1, 3], stream=True)
does your answer apply for this too? and is it recommended to use openCV with my camera instead of this line of code?

**Reply by RizwanMunawar** [reply]
@Zeynp62, your problem doesn't appear to be connected to the code; it seems to be related to the codec of the camera frames, which OpenCV is unable to process.
